pkg/segment/context/context.go

```diff
@@ -63,8 +65,28 @@ func propertiesFromContext(ctx context.Context) *properties {
 
 // setContextProperty sets the value of a key in given context for telemetry data
 func setContextProperty(ctx context.Context, key string, value interface{}) {
-	properties := propertiesFromContext(ctx)
-	if properties != nil {
-		properties.set(key, value)
+	cProperties := propertiesFromContext(ctx)
+	if cProperties != nil {
+		cProperties.set(key, value)
+	}
+}
+
+func SetClusterType(ctx context.Context, client *occlient.Client) {
+	var value string
+	if client == nil {
+		value = "not-found"
+	} else {
+		// We are not checking ServerVersion to decide the cluster type because it does not always return the version,
+		// it sometimes fails to retrieve the data if user is using minishift or plain oc cluster
+		if isOC, _ := client.IsProjectSupported(); isOC {
+			if isOC4, _ := client.IsOpenshift4(); isOC4 {
```

Comment:
Please handle the errors from these two calls. We should default to `kubernetes` etc only when there are no errors.

Reply:
Do you suggest we do error handling for  `if isOC4, _ := client.IsOpenshift4(); isOC4 {` as well?

Reply:
> Do you suggest we do error handling for if isOC4, _ := client.IsOpenshift4(); isOC4 { as well?

Yes